### PR TITLE
chore: Tidy up hierarchy from loose pool elements etc

### PR DESCRIPTION
**Pull request by fcolarich**
## What does this PR change?

In short, made it so most pools use the POOL_CONTAINER transform as a parent, so all are under that GO in the hierarchy. This removes a lot of visual clutter and also makes it easier to identify where and what pool elements are doing.

## How to test the changes?

Just do a fire test to make sure nothing broke 😁
The changes involve NFTs, Media Players (like the screens on the lobby), Name Tags, Other Avatars and Road segments, so it can all be seen on the Genesis Plaza (or not if it doesnt work xD)

Addresses #2319

### CODE REVIEW

File: Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs [hunk 1]
```diff
@@ -50,7 +51,8 @@ public RemoteEntities(
 
         public void Initialize(RemoteAvatarCollider remoteAvatarCollider)
         {
-            remoteAvatarColliderPool = componentPoolsRegistry.AddGameObjectPool(() => Object.Instantiate(remoteAvatarCollider, Vector3.zero, Quaternion.identity));
+            RemoteEntitiesParent = new GameObject("REMOTE_ENTITIES").transform;
```
Comment (m3taphysics) [inline Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs:54]: Does this need cleaning up in Dispose?

File: Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs [hunk 2]
```diff
@@ -125,7 +127,7 @@ private Entity CreateCharacter(RemoteProfile profile, World world)
         {
             var transform = transformPool.Get()!;
             transform.name = $"REMOTE_ENTITY_{profile.WalletId}";
-            transform.transform.SetParent(null);
+            transform.transform.SetParent(RemoteEntitiesParent);
```
Comment (m3taphysics) [inline Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs:130]: Should this only happen #ifdef UNITY_EDITOR for performance reasons?

**Review by m3taphysics — CHANGES_REQUESTED** [review]
Same comments for other locations do the parent objects need cleaning up? And we only want to parent when in editor right?

File: Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs (hunk 1 above)
Comment (fcolarich) [reply Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs:54]: Ahh you are right, we are not destroying these on Dispose!

File: Explorer/Assets/DCL/Roads/Pools/RoadAssetsPool.cs [hunk 3]
```diff
@@ -32,13 +34,18 @@ public class RoadAssetsPool : IRoadAssetPool, IDisposable
         /// </summary>
         private const int POOLS_MAX_NEW_INSTANCES = 20;
 
-        private readonly Transform roadAssetParent;
+        [CanBeNull] private readonly Transform roadAssetParent = null;
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Roads/Pools/RoadAssetsPool.cs:37]: Coz Rider said so

[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Roads/Pools/RoadAssetsPool.cs:37]: lemme try again again xD

File: Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs [hunk 4]
```diff
@@ -46,11 +47,14 @@ public RemoteEntities(
             this.componentPoolsRegistry = componentPoolsRegistry;
             this.queuePool = queuePool;
             this.collidersGlobalCache = collidersGlobalCache;
+#if UNITY_EDITOR
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Multiplayer/Profiles/Entities/RemoteEntities.cs:50]: We were discussing the hierarchy chaos the other day and Aga mentioned that in one of her projects they made it so only in editor they would reparent objects, as in build it doesnt matter how legible the hierarchy is and it creates (albeit very small) a footprint, particularly if the object is reparented many times later or if the object has a complex hierarchy itself. So by making it only happen on editor we gain on readability and on the other hand add a minimal boost to performance on build...

File: Explorer/Assets/DCL/Roads/Pools/RoadAssetsPool.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Roads/Pools/RoadAssetsPool.cs:37]: okay, Rider had some weird setting for this particular file... now I was able to 🤷

**Pull request merged**